src/inference.jl

```diff
@@ -508,7 +524,13 @@ function inference(;
         returnvars   = Dict(variable => returnoption for (variable, value) in pairs(vardict) if (israndom(value) && !isanonymous(value)))
     end
 
+    # Check if `predictvars` is nothing but `data` has missing values
+    if predictvars === nothing
+        predictvars = Dict(variable => KeepLast() for (variable, value) in pairs(vardict) if (isdata(value) && !isempty(findall(ismissing, data[variable])) && !isanonymous(value)))
```

Comment:
Perhaps `any(ismissing, data[variable])` should be more efficient than `!isempty(findall(ismissing, data[variable]))`.

Reply:
It occurs in `__inference_check_dataismissing` now.